shared/src/metabase/mbql/schema.cljc

```diff
@@ -476,7 +482,8 @@
 
 (def arithmetic-expressions
   "Set of valid arithmetic expression clause keywords."
-  #{:+ :- :/ :* :coalesce :length :round :ceil :floor :abs :power :sqrt :log :exp :case})
+  #{:+ :- :/ :* :coalesce :length :round :ceil :floor :abs :power :sqrt :log :exp :case
+    :datetime-diff})
```

Comment:
I saw that you had a question somewhere (on this PR I think) about whether `arithmetic-expressions` meant expressions whose arguments are numeric, or expressions whose results are numeric, and you got an answer. Please write that sort of stuff down in comments, or add it to the docstring! That way the next person who has that question will have the answer instead of them having to do the same thing you did

Reply:
Updated in https://github.com/metabase/metabase/pull/25722/commits/b9b0ef41fabd438c80f6fb310c50a88aee2e85c3. I've also left a bunch of TODOs in this commit https://github.com/metabase/metabase/pull/25722/commits/b9b0ef41fabd438c80f6fb310c50a88aee2e85c3 because I think we should rename `arithmetic-expressions` to `numeric-expressions` for clarity. There are also more functions that should be in `numeric/arithmetic-expressions` that aren't, e.g. temporal extraction functions like `get-year`. I'll leave it for a separate PR.